test/e2e-go/e2e_go_suite_test.go

```diff
@@ -0,0 +1,174 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e_go_test
+
+import (
+	"fmt"
+	"path"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo" //nolint:golint
+	. "github.com/onsi/gomega" //nolint:golint
+	kbtestutils "sigs.k8s.io/kubebuilder/test/e2e/utils"
+
+	testutils "github.com/operator-framework/operator-sdk/test/internal"
+)
+
+// TestE2EGo ensures the Go projects built with the SDK tool by using its binary.
+func TestE2EGo(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping Operator SDK E2E Go Suite testing in short mode")
+	}
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "E2EGo Suite")
+}
+
+var (
+	tc testutils.TestContext
+	// isPrometheusManagedBySuite is true when the suite tests is installing/uninstalling the Prometheus
+	isPrometheusManagedBySuite = true
+	// isOLMManagedBySuite is true when the suite tests is installing/uninstalling the OLM
+	isOLMManagedBySuite = true
+	// kubectx stores the k8s context from where the tests are running
+	kubectx string
+	// projectName is the name of the test project
+	projectName string
+)
+
+// BeforeSuite run before any specs are run to perform the required actions for all e2e Go tests.
+var _ = BeforeSuite(func(done Done) {
+	var err error
+
+	By("creating a new test context")
+	tc, err = testutils.NewTestContext("GO111MODULE=on")
+	Expect(err).NotTo(HaveOccurred())
+	Expect(tc.Prepare()).To(Succeed())
+	projectName = filepath.Base(tc.Dir)
+
+	By("checking the cluster type")
+	kubectx, err = tc.Kubectl.Command("config", "current-context")
+	Expect(err).Should(Succeed())
+
+	By("checking API resources applied on Cluster")
+	output, err := tc.Kubectl.Command("api-resources")
+	Expect(err).NotTo(HaveOccurred())
+	if strings.Contains(output, "servicemonitors") {
+		isPrometheusManagedBySuite = false
+	}
+	if strings.Contains(output, "clusterserviceversions") {
+		isOLMManagedBySuite = false
+	}
+
+	if isPrometheusManagedBySuite {
+		By("installing Prometheus")
+		Expect(tc.InstallPrometheusOperManager()).To(Succeed())
+
+		By("ensuring provisioned Prometheus Manager Service")
+		Eventually(func() error {
+			_, err := tc.Kubectl.Get(
+				false,
+				"Service", "prometheus-operator")
+			return err
+		}, 3*time.Minute, time.Second).Should(Succeed())
+	}
+
+	if isOLMManagedBySuite {
+		By("installing OLM")
+		Expect(tc.InstallOLM()).To(Succeed())
+	}
+
+	By("initializing a project")
+	projectName := filepath.Base(tc.Dir)
+	err = tc.Init(
+		"--project-version", "3-alpha",
+		"--repo", path.Join("github.com", "example", projectName),
+		"--domain", tc.Domain,
+		"--fetch-deps=false")
+	Expect(err).Should(Succeed())
+
+	// Todo: add to the init scaffold
+	By("by adding scorecard custom patch file")
+	err = tc.AddScorecardCustomPatchFile()
+	Expect(err).NotTo(HaveOccurred())
```

Comment:
Ditto from above. Also, I don't think we want to keep that comment around since the custom tests are just examples. We'll never scaffold a real scorecard config file with these examples present.

Reply:
It is fine we can remove the init. However, I understood that the idea was in a long term scaffold the examples by default. Then, it would remove the mock here as well. It is was the only reason for the comment.

Reply:
I think that idea was to add some boilerplate `kuttl` tests as part of the `create api` command.